db/migrate/20160204220108_create_addresses.rb

```diff
@@ -0,0 +1,16 @@
+class CreateAddresses < ActiveRecord::Migration
+  def change
+    create_table :addresses do |t|
+      t.integer :street_number
+      t.string :street_name
+      t.string :apartment
+      t.string :city
+      t.string :state
+      t.string :zip
+      t.references :address_type, index: true, foreign_key: true
```

Comment:
Very nice! As I reading this, I was worried I was going to see two address classes but you have this set up beautifully.